public/app/features/playlist/baseAPI.ts

```diff
@@ -0,0 +1,40 @@
+import { BaseQueryFn, createApi } from '@reduxjs/toolkit/query/react';
+import { lastValueFrom } from 'rxjs';
+
+import { BackendSrvRequest, config, getBackendSrv } from '@grafana/runtime';
+
+interface RequestOptions extends BackendSrvRequest {
+  manageError?: (err: unknown) => { error: unknown };
+  showErrorAlert?: boolean;
+
+  // rtk codegen sets this
+  body?: BackendSrvRequest['data'];
+}
+
+function createBackendSrvBaseQuery({ baseURL }: { baseURL: string }): BaseQueryFn<RequestOptions> {
+  async function backendSrvBaseQuery(requestOptions: RequestOptions) {
+    try {
+      const { data: responseData, ...meta } = await lastValueFrom(
+        getBackendSrv().fetch({
+          ...requestOptions,
+          url: baseURL + requestOptions.url,
+          showErrorAlert: requestOptions.showErrorAlert,
+          data: requestOptions.body,
+        })
+      );
+      return { data: responseData, meta };
+    } catch (error) {
+      return requestOptions.manageError ? requestOptions.manageError(error) : { error };
+    }
+  }
+
+  return backendSrvBaseQuery;
+}
+
+export const baseAPI = createApi({
+  reducerPath: 'playlistAPI',
+  baseQuery: createBackendSrvBaseQuery({
+    baseURL: `/apis/playlist.grafana.app/v0alpha1/namespaces/${config.namespace}/`,
+  }),
+  endpoints: () => ({}),
+});
```

Comment:
This file looks like it'll be quite similar for all clients, could it be also generated?